### PR TITLE
Introduce finished runs API

### PR DESCRIPTION
**Pull request by peregrineshahin**
This comes handy for Devs who want to list the finished runs without having to scrape webpages.

### CODE REVIEW

File: server/fishtest/api.py [hunk 1]
```diff
@@ -274,6 +274,46 @@ def active_runs(self):
...
+    @view_config(route_name="api_finished_runs")
+    def finished_runs(self):
+        self.__t0 = datetime.now(timezone.utc)
+        self.__api = "/api/finished_runs"
+
+        username = self.request.matchdict.get("username", "")
+        success_only = self.request.params.get("success_only", False)
+        yellow_only = self.request.params.get("yellow_only", False)
+        ltc_only = self.request.params.get("ltc_only", False)
+        timestamp = seld.request.params.get("timestamp", "")
```
Comment (ppigazzini) [inline server/fishtest/api.py:286]:
```diff
- timestamp = seld.request.params.get("timestamp", "")
+ timestamp = self.request.params.get("timestamp", "")
```

File: server/fishtest/api.py [hunk 2]
```diff
@@ -274,6 +274,46 @@ def active_runs(self):
...
+        page_idx = max(0, int(page_param) - 1) if page_param.isdigit() else 0
+        page_size = 50
+
+        last_updated = None
+        if timestamp != "" and re.match(r"^\d{10}(\.\d+)?$", timestamp):
+            last_updated = datetime.fromtimestamp(float(timestamp))
+        elif timestamp != "":
+            self.handle_error("Please provide a valid UNIX timestamp.")
+
+        runs, num_finished = request.rundb.get_finished_runs(
```
Comment (ppigazzini) [inline server/fishtest/api.py:298]:
```diff
- runs, num_finished = request.rundb.get_finished_runs(
+ runs, num_finished = self.request.rundb.get_finished_runs(
```

**Comment by ppigazzini** [comment]
DEV updated with the bug fixes:
https://dfts-0.pigazzini.it/api/finished_runs

**Comment by ppigazzini** [comment]
DEV updated.

**Comment by peregrineshahin** [comment]
Looks good I think

**Comment by ppigazzini** [comment]
https://dfts-0.pigazzini.it/api/finished_runs?page=2927
https://dfts-0.pigazzini.it/api/finished_runs?page=2&username=Peregrine

**Comment by ppigazzini** [comment]
Looks good, I0m not an expert of the usage of this information, so I wait if there are suggestions by @vondele

**Comment by vondele** [comment]
looks good to me.

**Review by ppigazzini — APPROVED** [review]

**Comment by ppigazzini** [comment]
PROD updated, thank you @peregrineshahin :)

**Pull request merged**